### PR TITLE
cloudcompare: creation of .desktop entry

### PR DESCRIPTION
**Pull request by locnide**
CloudCompare no longer provides snap directory with images and .desktop entry since they removed snap builds support. https://github.com/CloudCompare/CloudCompare/commit/6e6e3f13d76e5bf44a3938642e8035942d994d26
Those files where used during the nix package build to create a .desktop entry. Since they no longer exists the build fails : https://hydra.nixos.org/build/251747046/nixlog/1/tail

## Description of changes

To solve the issue, the build of the desktop entry added in [efdec26090fe6c61327b67d933e39d694da36bc2](https://github.com/NixOS/nixpkgs/commit/efdec26090fe6c61327b67d933e39d694da36bc2#diff-6806a6534f483e9dc42cf06ae8e134523a589d38f7a8621a43db9c2cc4258de2) has been removed.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — APPROVED** [review]

**Review by drupol — CHANGES_REQUESTED** [review]
I agree with the PR, but to avoid breaking the desktop environment and confusing users relying on `.desktop` files, would it be possible to create the desktop entry within this PR ? It's quite trivial to do.

**Comment by koalp** [comment]
> I agree with the PR, but to avoid breaking the desktop environment and confusing users relying on `.desktop` files, would it be possible to create the desktop entry within this PR ? It's quite trivial to do.

You're right. I removed it because the .desktop entries were introduced in unstable
I put them back.

File: pkgs/applications/graphics/cloudcompare/default.nix [hunk 1]
```diff
@@ -96,19 +99,45 @@ mkDerivation rec {
...
+      comment = "3D point cloud and mesh processing software";
+      exec = "ccViewer";
+      terminal = false;
+      categories = [ "Graphics" "3DGraphics" "Viewer" ];
+      keywords = [ "3d" "viewer" ];
+      icon = "ccViewer";
+    })
+  ];
+
   meta = with lib; {
```
Comment (drupol) [inline pkgs/applications/graphics/cloudcompare/default.nix:141]: Could you add missing meta.mainProgram ?

**Review by drupol — APPROVED** [review]

**Review by drupol — CHANGES_REQUESTED** [review]
Looks like it's not building anymore.

**Comment by koalp** [comment]
> Looks like it's not building anymore.

Seems that [default jdk version](https://github.com/NixOS/nixpkgs/pull/273811) has been changed 2 days ago and gdal dependency do not build anymore.
I will look at it later, either by using gdalMinimal if it is possible, or fixing gdal.

**Comment by drupol** [comment]
I've set the PR in draft in the meantime.

**Review by drupol — APPROVED** [review]

**Comment by koalp** [comment]
The issue have been solved in gdal. Cloudcompare builds again.
I rebased on master.

I just saw that it was updated to v2.13.1. I updated the package. Sorry for the noise.

**Pull request merged**